include/deal.II/lac/generic_linear_algebra.h

```diff
@@ -68,10 +68,51 @@ namespace LinearAlgebraPETSc
 {
   using namespace dealii;
 
+  /**
+   * Typedef for the base class for preconditioners.
+   */
+  typedef PETScWrappers::PreconditionerBase PreconditionerBase;
+
+  /**
+   * Typedef for the base class for linear solvers.
+   */
+  typedef PETScWrappers::SolverBase SolverBase;
```

Comment:
This is so that one can easily switch between the solvers and preconditioners. An example:
```c++
std_cxx11::unique_ptr<LA::SolverBase> solver;
std_cxx11::unique_ptr<LA::PreconditionerBase> preconditioner;

if (parameters.type_lin == "CG") {
  solver.reset(new LA::SolverCG(solver_control));
} ...

if (parameters.preconditioner_type == "Jacobi") {
  LA::MPI::PreconditionJacobi* ptr_prec
    = new LA::MPI::PreconditionJacobi ();

  LA::MPI::PreconditionJacobi::AdditionalData
    additional_data (parameters.preconditioner_relaxation);

  ptr_prec->initialize(tangent_matrix.block(u_block,u_block), additional_data);
  preconditioner.reset(ptr_prec);
} ...
```